snuba/query/processors/handled_functions.py

```diff
@@ -0,0 +1,95 @@
+from snuba.query.expressions import (
+    Argument,
+    Expression,
+    FunctionCall,
+    Lambda,
+    Literal,
+)
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.query.conditions import (
+    BooleanFunctions,
+    ConditionFunctions,
+    binary_condition,
+)
+from snuba.request.request_settings import RequestSettings
+
+
+class HandledFunctionsProcessor(QueryProcessor):
+    """
+    Adds the isHandled and notHandled snuba functions.
+
+    The implementation of these functions is too complex for clients to provide so
+    these wrappers are required.
+
+    - The `isHandled` function searches an array field for null or 1.
+    - The `notHandled` function searches an array field for 0, null
+      values will be excluded from the result.
+
+    Both functions return 1 or 0 if a row matches.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_functions(exp: Expression) -> Expression:
+            if isinstance(exp, FunctionCall):
+                if exp.function_name == "isHandled":
+                    return FunctionCall(
```

Comment:
Could you please validate the function call? For example check it has the right number of parameters ? If it is not valid please raise any `snuba.query.exceptions.InvalidExpressionException`. That will trigger an HTTP400 instead of 500.

Reply:
Done, I used the same signature validator that was used in CustomFunction.